diff -ruN original/etc/init.d/S12UpdateRFHardware  current/etc/init.d/S12UpdateRFHardware
--- original/etc/init.d/S12UpdateRFHardware
+++ current/etc/init.d/S12UpdateRFHardware
@@ -89,13 +89,13 @@

       echo -n "RPI-RF-MOD: "

-      NEW_VERSION=$(ls /firmware/RPI-RF-MOD/dualcopro_update_blhmip-*.eq3 | sed 's/.*-\(.*\)\.eq3/\1/')
+      NEW_VERSION=$(ls /firmware/RPI-RF-MOD/dualcopro_update_blhmip-*.eq3 | tail -n1 | sed 's/.*-\(.*\)\.eq3/\1/')
       OLD_VERSION=$(/opt/java/bin/java -Dgnu.io.rxtx.SerialPorts=${HM_HMIP_DEVNODE} -jar /opt/HmIP/hmip-copro-update.jar -p ${HM_HMIP_DEVNODE} -v 2>/dev/null | grep "Application version =" | cut -d' ' -f5)
       if [[ "${NEW_VERSION}" != "${OLD_VERSION}" ]]; then
         echo -n "${OLD_VERSION}=>${NEW_VERSION}, "

         # run firmware update
-        HM_HMIP_VERSION=$(/opt/java/bin/java -Dgnu.io.rxtx.SerialPorts=${HM_HMIP_DEVNODE} -jar /opt/HmIP/hmip-copro-update.jar -p ${HM_HMIP_DEVNODE} -o -f /firmware/RPI-RF-MOD/dualcopro_update_blhmip-*.eq3 2>/dev/null | grep "Version:" | cut -d' ' -f5)
+        HM_HMIP_VERSION=$(/opt/java/bin/java -Dgnu.io.rxtx.SerialPorts=${HM_HMIP_DEVNODE} -jar /opt/HmIP/hmip-copro-update.jar -p ${HM_HMIP_DEVNODE} -o -f /firmware/RPI-RF-MOD/dualcopro_update_blhmip-${NEW_VERSION}.eq3 2>/dev/null | grep "Version:" | cut -d' ' -f5)

         echo "${HM_HMIP_VERSION}" >/var/hmip_firmware_version
         if [[ -z "${HM_HMRF_DEV}" ]]; then
